### PR TITLE
Refactor pinot-connectors to break the dependencies from kafka 0.9

### PR DESCRIPTION
**Pull request by fx19880617**
Currently `pinot-tools`/`pinot-integration-tests`/`pinot-perf` modules have compile dependency on `pinot-connector-kafka-0.9`, which prevents us to add new connectors.

This PR extracts common interfaces of StreamDataServer and StreamDataProducer. And create those instance based on runtime class name.

So we could make kafka-connector as a runtime dependency for `pinot-tools`/`pinot-integration-tests`/`pinot-perf`.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/StreamDataServerStartable.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+package org.apache.pinot.core.realtime.stream;
+
+import java.util.Properties;
+
+
+/**
+ * StreamDataServerStartable is the interface for stream data sources. E.g. KafkaDataServerStartable, KinesisDataServerStarable.
+ */
+public interface StreamDataServerStartable {
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/StreamDataServerStartable.java]: simple java docs for this.

**Comment by fx19880617** [comment]
I've tested hybrid and realtime quick start. Both of them work.

File: pinot-tools/src/main/java/org/apache/pinot/tools/KafkaStarterUtils.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  public static int DEFAULT_KAFKA_PORT = 19092;
+  public static final String DEFAULT_KAFKA_BROKER = "localhost:" + DEFAULT_KAFKA_PORT;
+
+  public static final String PORT = "port";
+  public static final String BROKER_ID = "broker.id";
+  private static final String ZOOKEEPER_CONNECT = "zookeeper.connect";
+  private static final String LOG_DIRS = "log.dirs";
+
+  public static final String KAFKA_09_SERVER_STARTABLE_CLASS_NAME =
+      "org.apache.pinot.core.realtime.impl.kafka.server.KafkaDataServerStartable";
```
Comment (npawar) [inline pinot-tools/src/main/java/org/apache/pinot/tools/KafkaStarterUtils.java]: is the expectation now that every kafka implementation should have the KafkaDataServerStartable in the path `org.apache.pinot.core.realtime.impl.kafka.server` if we want the pinot-tools/pinot-perf/pinot-integration-tests to run?

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/StreamAvroIntoKafkaCommand.java [hunk 3]
```diff
@@ -104,8 +102,12 @@ public boolean execute()
     properties.put("serializer.class", "kafka.serializer.DefaultEncoder");
     properties.put("request.required.acks", "1");
 
-    ProducerConfig producerConfig = new ProducerConfig(properties);
-    Producer<byte[], byte[]> producer = new Producer<byte[], byte[]>(producerConfig);
+    StreamDataProducer streamDataProducer;
+    try {
+      streamDataProducer = StreamDataProvider.getStreamDataProducer(KafkaStarterUtils.KAFKA_09_PRODUCER_CLASS_NAME, properties);
```
Comment (npawar) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/StreamAvroIntoKafkaCommand.java]: why keep `09` in the name? I thought these same files are going to be able to handle other versions as well?

**Comment by npawar** [comment]
Minor questions, LGTM otherwise. Thanks a lot for doing this!

File: pinot-tools/src/main/java/org/apache/pinot/tools/KafkaStarterUtils.java (hunk 2 above)
Comment (kishoreg) [reply pinot-tools/src/main/java/org/apache/pinot/tools/KafkaStarterUtils.java]: yes, we can probably use profiles to add the right dependency? https://stackoverflow.com/questions/166895/different-dependencies-for-different-build-profiles

Comment (fx19880617) [reply pinot-tools/src/main/java/org/apache/pinot/tools/KafkaStarterUtils.java]: I think if we want to have an integration test, then data server/provider has to be implemented. Either we can have a mock one or we directly wrapper a real implementation.

**Pull request merged**